File: app/src/main/java/me/manabu/activities/LessonCardActivity.java
package me.manabu.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.widget.Adapter;
import android.widget.ListView;
import android.widget.TextView;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.List;

import me.manabu.R;
import me.manabu.adapters.LessonItemAdapter;
import me.manabu.adapters.models.LessonItemModel;
import me.manabu.api.models.DeckModel;
import me.manabu.helpers.RetrofitHelper;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LessonCardActivity extends AppCompatActivity {

    private int DECK_ID;

    private TextView itemName;
    private ListView listViewInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lesson_card);

        itemName = (TextView) findViewById(R.id.lesson_item_name);
        listViewInfo = (ListView) findViewById(R.id.lesson_listview_info);

        Toolbar toolbar = (Toolbar) findViewById(R.id.lesson_toolbar_include);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        Intent intent = getIntent();
        DECK_ID = intent.getIntExtra("deckId", 0);

        loadData();
    }

    private void loadData(){
        RetrofitHelper.retrofit.getLessonsFromDeck(DECK_ID).enqueue(new Callback<List<DeckModel>>() {
            @Override
            public void onResponse(Call<List<DeckModel>> call, Response<List<DeckModel>> response) {
                if(response.isSuccessful()){
                    Log.d("LessonCardActivity", "Got deck #" + DECK_ID + "!");

                    DeckModel deck = response.body().get(0);
                    fillData(deck);
                } else {
                    Log.d("LessonCardActivity", "Can't get deck for some reason.");
                }
            }

            @Override
            public void onFailure(Call<List<DeckModel>> call, Throwable t) {
                Log.d("LessonCardActivity", "Get deck failed - " + t.toString());
            }
        });
    }

    private void fillData(DeckModel deck){
        getSupportActionBar().setTitle(deck.getName());
        itemName.setText(deck.getId1().getTarget());
        ArrayList<LessonItemModel> infoList = new ArrayList<>();

        //TODO: To strings
        infoList.add(new LessonItemModel("Desc", deck.getId1().getDescription()));
        infoList.add(new LessonItemModel("Mnemonic", deck.getId1().getMnemonic()));

        LessonItemAdapter adapter = new LessonItemAdapter(this, infoList);

        listViewInfo.setAdapter(adapter);
        //descriptionText.setText(deck.getId1().getDescription());
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/me/manabu/activities/LessonCardActivity.kt
package me.manabu.activities

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import kotlinx.android.synthetic.main.activity_lessons.*
import me.manabu.R
import me.manabu.activities.fragments.LessonCardFragment

class LessonCardActivity : FragmentActivity()  {

    //private var DECK_ID: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lessons)
        //setToolbar(R.id.lessonToolbarInclude)

//        val intent = intent
//        DECK_ID = intent.getIntExtra("deckId", 0)

        //loadData()

        ////////////////

        val test = arrayOf("rofl1", "rofl2", "rofl3")

        val pager = lessonCardsPager
        val pagerAdapter = ScreenSlidePagerAdapter(supportFragmentManager, test)
        pager.adapter = pagerAdapter



    }

    private class ScreenSlidePagerAdapter(fm: FragmentManager, var test: Array<String>) : FragmentStatePagerAdapter(fm) {
        override fun getItem(position: Int): Fragment {
            return LessonCardFragment()
        }

        override fun getCount(): Int {
            return test.size
        }
    }

//
//    private fun loadData() {
//        Api.retrofit.getLessonsFromDeck(DECK_ID).enqueue(object : Callback<List<DeckModel>> {
//            override fun onResponse(call: Call<List<DeckModel>>, response: Response<List<DeckModel>>) {
//                if (response.isSuccessful) {
//                    Log.d("LessonCardActivity", "Got deck #$DECK_ID!")
//
//                    val deck = response.body()!![0]
//                    fillData(deck)
//                } else {
//                    Log.d("LessonCardActivity", "Can't get deck for some reason.")
//                }
//            }
//
//            override fun onFailure(call: Call<List<DeckModel>>, t: Throwable) {
//                Log.d("LessonCardActivity", "Get deck failed - " + t.toString())
//            }
//        })
//    }
//
//    private fun fillData(deck: DeckModel) {
//        supportActionBar!!.title = deck.name
//        lessonItemName.text = deck.id1.target
//        val infoList = ArrayList<LessonItemModel>()
//
//        //TODO: To strings
//        infoList.add(LessonItemModel("Desc", deck.id1.description))
//        infoList.add(LessonItemModel("Mnemonic", deck.id1.mnemonic))
//
//        val adapter = LessonItemAdapter(this, infoList)
//        lessonListview.adapter = adapter
//        //descriptionText.setText(deck.getId1().getDescription());
//    }

//    override fun onSupportNavigateUp(): Boolean {
//        onBackPressed()
//        return true
//    }
}